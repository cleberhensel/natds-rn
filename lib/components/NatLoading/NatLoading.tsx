import React from "react";
import { ActivityIndicator } from "react-native";
import { colors } from "../../assets/theme/colors/colors";
import { LoadingWrapper } from "./NatLoading.styles";
export const NatLoading = () => (
  <LoadingWrapper>
    <ActivityIndicator size="large" color={colors.primary.orange} />
  </LoadingWrapper>
);
