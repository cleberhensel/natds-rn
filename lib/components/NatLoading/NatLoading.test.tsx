import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components/native";
import { NatLoading } from "./NatLoading";
describe("Container component", () => {
  it("Should render loading component correctly", () => {
    const container = renderer.create(<NatLoading />).toJSON();
    expect(container).toMatchSnapshot();
  });
  it("Should render loading component with static styling", () => {
    const loading = renderer.create(<NatLoading />).toJSON();
    expect(loading).toHaveStyleRule("position", "absolute");
    expect(loading).toHaveStyleRule("justify-content", "center");
    expect(loading).toHaveStyleRule("align-items", "center");
    expect(loading).toHaveStyleRule("right", 0);
    expect(loading).toHaveStyleRule("left", 0);
    expect(loading).toHaveStyleRule("top", 0);
    expect(loading).toHaveStyleRule("bottom", 0);
    expect(loading).toHaveStyleRule("z-index", 10);
  });
});
