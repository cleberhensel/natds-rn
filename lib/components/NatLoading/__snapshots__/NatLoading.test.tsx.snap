// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component Should render loading component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#FF6B0B"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
